File: 121. Best Time to Buy and Sell Stock/sol.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minvalue = INT_MAX;
        int maxprofit = 0;
        for(int i=0;i< prices.size();i++){
            if(minvalue > prices[i]){minvalue = prices[i];}
            else if (maxprofit < (prices[i] - minvalue)){maxprofit = prices[i] - minvalue;}
            
        }
        
        return maxprofit;
    }
        
};

File: 121. Best Time to Buy and Sell Stock/sol.py
class Solution:
    def maxProfit(self, prices):
        
        mintili = inf
        profit = 0
        
        for i in range(len(prices)):
            
            # profit we will get assumes sell at day i
            profit = max(profit, prices[i] - mintili)
            
            # update the best timing to buy stocks
            mintili = min(mintili, prices[i])
            
        return profit
        
